src/daft-io/src/utils.rs

```diff
@@ -0,0 +1,29 @@
+use url::{ParseError, Position};
+
+use crate::Error;
+
+pub(crate) const DELIMITER: &str = "/";
+pub fn parse_object_url(uri: &str) -> super::Result<(String, String, String)> {
```

Comment:
Let's add some docstrings and also perhaps return a struct instead of this tuple so that it's more clear what the three strings are.